src/FakeItEasy/Creation/DelegateProxies/CastleInvocationCallAdapter.cs

```diff
@@ -0,0 +1,98 @@
+namespace FakeItEasy.Creation.DelegateProxies
+{
+    using System;
+    using System.Diagnostics;
+    using System.Reflection;
+    using Castle.DynamicProxy;
+    using FakeItEasy.Configuration;
+    using FakeItEasy.Core;
+
+    /// <summary>
+    /// An adapter that adapts an <see cref="IInvocation" /> to a <see cref="IFakeObjectCall" />.
+    /// </summary>
+    internal class CastleInvocationCallAdapter
+        : InterceptedFakeObjectCall
+    {
+        private readonly IInvocation invocation;
+        private readonly Delegate theDelegate;
+        private readonly object[] originalArguments;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CastleInvocationCallAdapter"/> class.
+        /// </summary>
+        /// <param name="invocation">The invocation.</param>
+        /// <param name="theDelegate">The delegate teh invocation was made on.</param>
+        [DebuggerStepThrough]
+        public CastleInvocationCallAdapter(IInvocation invocation, Delegate theDelegate)
+        {
+            this.invocation = invocation;
+            this.theDelegate = theDelegate;
+            var savedArguments = new object[invocation.Arguments.Length];
+            Array.Copy(invocation.Arguments, savedArguments, savedArguments.Length);
+            this.originalArguments = savedArguments;
+            this.Method = invocation.Method;
+            this.Arguments = new ArgumentCollection(invocation.Arguments, invocation.Method);
+        }
+
+        /// <summary>
+        /// Gets the method that's called.
+        /// </summary>
+        public override MethodInfo Method { get; }
+
+        /// <summary>
+        /// Gets the arguments used in the call.
+        /// </summary>
+        public override ArgumentCollection Arguments { get; }
+
+        /// <summary>
+        /// Gets the faked object the call is performed on.
+        /// </summary>
+        public override object FakedObject => this.theDelegate;
+
+        public override object? ReturnValue
+        {
+            get => this.invocation.ReturnValue;
+            set => this.invocation.ReturnValue = value;
+        }
+
+        /// <summary>
+        /// Returns a completed call suitable for being recorded.
+        /// </summary>
+        /// <returns>A completed fake object call.</returns>
+        public override CompletedFakeObjectCall ToCompletedCall()
+        {
+            return new CompletedFakeObjectCall(
+                this,
+                this.originalArguments);
+        }
+
+        /// <summary>
+        /// Calls the base method, should not be used with interface types.
+        /// </summary>
+        public override void CallBaseMethod()
+        {
+            this.invocation.Proceed();

```

Comment:
This is an error. We should be throwing an exception. Looks like we're missing a spec!

Reply:
Except we're not? https://github.com/FakeItEasy/FakeItEasy/blob/2a8b4230e9f7bb1587f659c85619d3bd29b6914a/tests/FakeItEasy.Specs/FakingDelegates.cs#L97

Reply:
Oh. We catch it at configuration time!
https://github.com/FakeItEasy/FakeItEasy/blob/2a8b4230e9f7bb1587f659c85619d3bd29b6914a/src/FakeItEasy/Configuration/RuleBuilder.cs#L108

Still. I don't _love_ the proceed here. I'll restore `throw new FakeConfigurationException(ExceptionMessages.DelegateCannotCallBaseMethod)`. Or something else if you'd rather.

Reply:
Sounds good to me!